src/statement.ts

```diff
@@ -0,0 +1,240 @@
+            const waited = await receipt.wait();
+            receipt = { ...receipt, ...waited };
+          }
+
+          return wrapResult(receipt, performance.now() - start);
+        }
+      }
+    } catch (cause) {
+      if (
+        cause instanceof Error &&
+        cause.message.includes("column not found")
+      ) {
+        throw new Error("COLUMN_NOTFOUND", { cause });
+      }
+      throw new Error("ALL_ERROR", { cause });
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async first<T = S, K extends keyof T = keyof T>(): Promise<T>;
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async first<T = S, K extends keyof T = keyof T>(
+    colName: undefined,
+    opts?: Signal
+  ): Promise<T>;
+  async first<T = S, K extends keyof T = keyof T>(
+    colName: K,
+    opts?: Signal
+  ): Promise<T[K] | null>;
+  async first<T = S, K extends keyof T = keyof T>(
+    colName?: K,
+    opts: Signal = {}
+  ): Promise<T | T[K] | null> {
+    try {
+      const { sql, type, tables } = await this.#parseAndExtract();
+      switch (type) {
+        case "read": {
+          const config = await extractReadonly(this.config, {
+            type,
+            tables,
+          });
+          const results = await queryFirst<T>(config, sql, opts);
+          if (results == null || colName == null) {
+            return results;
+          }
+          return extractColumn(results, colName);
+        }
+        default: {
+          const receipt = await exec(this.config, { type, sql, tables });
+          /* c8 ignore next */
+          if (this.config.autoWait ?? false) {
+            await receipt.wait();
+          }
+          return null;
+        }
+      }
+    } catch (cause) {
+      if (
+        cause instanceof Error &&
+        cause.message.includes("column not found")
+      ) {
+        throw new Error("COLUMN_NOTFOUND", { cause });
+      }
+      throw new Error("FIRST_ERROR", { cause });
+    }
+  }
+
+  async run(opts: Signal = {}): Promise<Result<never>> {
+    try {
+      const start = performance.now();
+      const { sql, type, tables } = await this.#parseAndExtract();
+      switch (type) {
+        case "read": {
+          const config = await extractReadonly(this.config, {
+            type,
+            tables,
+          });
+          await queryAll(config, sql, opts);
+          return wrapResult([], performance.now() - start);
+        }
+        default: {
+          let receipt = await exec(this.config, { type, sql, tables });
+          if (this.config.autoWait ?? false) {
+            const waited = await receipt.wait();
+            receipt = { ...receipt, ...waited };
+          }
+          return wrapResult(receipt, performance.now() - start);
+        }
+      }
+    } catch (cause) {
+      throw new Error("RUN_ERROR", { cause });
+    }
+  }
+
+  async raw<T = S>(opts: Signal = {}): Promise<Array<ValueOf<T>>> {
```

Comment:
same as "all", but only returns raw rows (which uses format=table) in the validator api call.

Reply:
putting a note here for my own reference. example response is:
```
await db.prepare(`SELECT * FROM ${queryableName};`).raw();
// => [ [ 'tree', 'aspen' ], [ 'tree', 'pine' ] ]
```

Reply:
Yes, raw is essentially what our current "table" version returns, minus the extra columns field with the column names.